themes/src/main/resources/theme/keycloak.v2/login/template.ftl

```diff
@@ -48,6 +48,22 @@
         checkCookiesAndSetTimer(
             "${url.ssoLoginInOtherTabsUrl?no_esc}"
         );
+
+        const DARK_MODE_CLASS = "pf-v5-theme-dark";
+        const mediaQuery =
+          window.matchMedia && window.matchMedia("(prefers-color-scheme: dark)");
+        updateDarkMode(mediaQuery?.matches);
+        mediaQuery?.addEventListener("change", (event) =>
+          updateDarkMode(event.matches),
+        );
+        function updateDarkMode(isEnabled = false) {
```

Comment:
No need to feature detect `matchMedia()` it is a [well supported API](https://developer.mozilla.org/en-US/docs/Web/API/Window/devicePixelRatio#browser_compatibility).

```suggestion
        const mediaQuery =window.matchMedia("(prefers-color-scheme: dark)");
        updateDarkMode(mediaQuery.matches);
        mediaQuery.addEventListener("change", (event) =>
          updateDarkMode(event.matches),
        );
        function updateDarkMode(isEnabled) {
```

Reply:
You would think that it wouldn't be needed, but I had issues with it on some browsers